### PR TITLE
Reformatted and improved Readme.md

### PR DESCRIPTION
**Pull request by mgreter**
Our main readme was getting pretty dusty and was missing a few new documentation links.

Decided to reformatted the whole thing, fixed quite a few typos and also added a few paragraphs here and there. Feel free to nitpick my english and to add more corrections. I also added two new badges from http://isitmaintained.com/. IMO it [looks and reads a lot better already][1].

[1]: https://github.com/mgreter/libsass/blob/readme/Readme.md

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,95 +1,97 @@
...
-While LibSass is a library primarily written in C++, it provides a simple
-C interface which should be used by most implementers. LibSass does not do
-much on its own without an implementer. This can be a command line tool, like
-[sassc](https://github.com/sass/sassc) or a [binding](docs/implementations.md)
-to your favorite programing language.
-
-If you want to build or interface with LibSass, we recommend to check out the
-documentation pages about [building LibSass](docs/build.md) and
-the [C-API documentation](docs/api-doc.md).
+Current LibSass 3.4 should be compatible with Sass 3.4. Please refer to the [sass compatibility
```
Comment (nschonni) [inline Readme.md]: "3.4 branch" with a link?

File: Readme.md [hunk 2]
```diff
@@ -1,95 +1,97 @@
...
-and tests there.
-
-We even run Travis tests for SassC!
-
-Tests
+Testing
 -------
 
-Since LibSass is a pure library, tests are run through the [SassSpec](https://github.com/sass/sass-spec) project using the [SassC](http://github.com/sass/sassc) driver.
+Since LibSass is a pure library, tests are run through the [Sass-Spec](https://github.com/sass/sass-spec)
```
Comment (nschonni) [inline Readme.md]: Think "sass-spec" is always lowercase

File: Readme.md [hunk 3]
```diff
@@ -1,95 +1,97 @@
...
+developing, you can run `./script/spec`. This will clone SassC and Sass-Spec under the project folder and
+then run the Sass-Spec test suite. You may want to update the clones to ensure you have the latest version.
+Note that the scripts in the `./script` folder are mainly intended for our CI needs.
 
-To run tests against LibSass while developing, you can run `./script/spec`. This will clone SassC and Sass-Spec under the project folder and then run the Sass-Spec test suite. You may want to update the clones to ensure you have the latest version.
+Building
+--------
 
-### DEBUG builds
+To build LibSass you need GCC 4.6+ or Clang/LLVM. If your OS is older, you may need to upgrade
```
Comment (nschonni) [inline Readme.md]: Is there a minimum Clang/LLVM version too to call out?

File: Readme.md [hunk 4]
```diff
@@ -1,95 +1,97 @@
...
 About
 -----
 
-LibSass is a C/C++ port of the Sass CSS precompiler. The original version was written in Ruby, but this version is meant for efficiency and portability.
-
-This library strives to be light, simple, and easy to build and integrate with a variety of platforms and languages.
+LibSass is a C++ port of the original Ruby Sass CSS compiler with a [C API](docs/api-doc.md).
+We coded LibSass with portability and efficiency in mind. You can expect LibSass to be magnitudes
+faster than Ruby Sass. We would even go as far and claim that it is the fastest CSS compiler
+around, which should be the case given it's written in C++.
```
Comment (xzyfer) [inline Readme.md]: >You can expect LibSass to be magnitudes faster than Ruby Sass. We would even go as far and claim that it is the fastest CSS compiler around, which should be the case given it's written in C++.

This bragging is unnecessary.

File: Readme.md [hunk 5]
```diff
@@ -1,95 +1,97 @@
...
 
 Developing
 ----------
 
-As you may have noticed, the LibSass repo itself has
-no executables and no tests. Oh noes! How can you develop???
+As noted above, the LibSass repository does not contain any binaries or other way to execute
```
Comment (xzyfer) [inline Readme.md]: I'm struggling to follow this paragraph. IMHO the original was good.

File: Readme.md (hunk 3 above)
Comment (xzyfer) [inline Readme.md]: With clang it's hard to know. If you're on OSX (which is where clang is prevalent) it'll just work.

File: Readme.md (hunk 1 above)
Comment (xzyfer) [inline Readme.md]: The 3.4 branch isn't an official thing. It was an implementation detail because I wasn't ready to drop some large changes into Node Sass. Node Sass now has the latest 3.5 betas on master.

Comment (xzyfer) [inline Readme.md]: Additionally I believe http://sass-compatibility.github.io/  is no longer actively maintained. I did the last round of updates a long time ago.

We should just say that we aim to be compatible the latest Ruby Sass stable release. All known incompatibilities are in our issue tracker.

File: Readme.md [hunk 6]
```diff
@@ -1,95 +1,97 @@
-LibSass
-=======
+LibSass - Sass compiler written in C++
+======================================
 
 by Aaron Leung ([@akhleung]), Hampton Catlin ([@hcatlin]), Marcel Greter ([@mgreter]) and Michael Mifsud ([@xzyfer])
```
Comment (xzyfer) [inline Readme.md]: It's worth distinguishing current maintainers. People still contact Hampton directly who then proxies messages on to me.

File: Readme.md (hunk 2 above)
Comment (xzyfer) [inline Readme.md]: Sass Spec is fine. The "Sass" capitalization is part of the Sass branding.

**Comment by xzyfer** [comment]
Nice work on the update. I found the language a little hard to follow in some places. I've left comments.

File: Readme.md [hunk 7]
```diff
@@ -1,95 +1,97 @@
...
+[SassC](https://github.com/sass/sassc), which is the official lightweight
+CLI tool built by the same people as LibSass.
+
+### Excerpt of "sanctioned" implementations:
+
+- https://github.com/sass/node-sass (Node.js)
+- https://github.com/sass/perl-libsass (Perl)
+- https://github.com/wellington/go-libsass (Go)
+- https://github.com/sass/sassc-ruby (Ruby)
+- https://github.com/medialize/sass.js (JS)
```
Comment (xzyfer) [inline Readme.md]: https://github.com/sass/libsass-net is now under the Sass org also.

File: Readme.md (hunk 4 above)
Comment (mgreter) [inline Readme.md]: Can you provide an alternative paragraph?

File: Readme.md (hunk 1 above)
Comment (mgreter) [inline Readme.md]: Can you provide an alternative paragraph?

File: Readme.md (hunk 5 above)
Comment (mgreter) [inline Readme.md]: Not sure what you mean exactly? IMO it adds a bit more information than the old wording. And I don't think the old wording was any less confusing? IMO the whole paragraph does not make much sense, but I've left it in since it already was there.

File: Readme.md (hunk 6 above)
Comment (mgreter) [inline Readme.md]: Can you provide an alternative paragraph?

File: Readme.md (hunk 7 above)
Comment (mgreter) [inline Readme.md]: I included the url

File: Readme.md (hunk 6 above)
Comment (mgreter) [inline Readme.md]: I separated it into two lines with "current mainters" and "original creators"

File: Readme.md (hunk 4 above)
Comment (mgreter) [inline Readme.md]: Tried the rewrite it a little and be more "humble". I really would like to keep the emphasize on the speed, because that is IMO one of the core and key assets of libsass.

**Comment by mgreter** [comment]
I've updated some parts a bit. Please provide alternative wording or paragraphs for the open things. Otherwise it's difficult to improve it without specific feedback. Thanks!

**Comment by xzyfer** [comment]
I'll provide some copy this week.

**Comment by mgreter** [comment]
@xzyfer still not heard back from you on this one!

**Comment by xzyfer** [comment]
Sorry I've been preoccupied with tracking that Alpine issues since it's blocked 3.5.0.beta.3. Feel free to ship this if you want and I'll PR fix later. My priority atm is figuring out what's causing the memory issues with Alpine.

**Comment by mgreter** [comment]
OK. Btw. I suspect a compiler (buggy toolchain) issue on alpine linux. I debugged it down to the heisenbug on the assembler level, but couldn't see why accessing a function jmp fails after the same jmp/function was called successfully before many times at the same memory location. I verified that the memory location is not overwritten. So for me this is a "wont fix" until we see similar issues on other platforms!

**Comment by mgreter** [comment]
FWIW. The only difference between node-sass and sassc is that node-sass seems to flag some memory pages as non-executable (as seen when running with strace), which could be a reason why the jmp could fail. But I wasn't able to really make a connection (the raw memory addressess don't seem to overlap).

**Comment by mgreter** [comment]
And in the end musl libc is a very young libc implementation (initial release is less than 5 years old, so chances are that it is simply a bug in said musl libc implementation) ...

**Comment by xzyfer** [comment]
I suspect you're correct. However if there is something we can do to mitigate the issue I'd like to try. Alpine is becoming a very important tool within the containerisation realm.
